yaksh/evaluator_tests/test_java_evaluation.py

```diff
@@ -507,5 +507,334 @@ class Test{
+                                    proc = subprocess.Popen("{}".format(cmd),
+                                                            shell=True,
+                                                            stdout=subprocess.PIPE,
+                                                            stderr=subprocess.PIPE
+                                                            )
+                                    stdout,stderr = proc.communicate()
+                                    return stdout,stderr
+                                cmds = ["javac Test.java", "java Test"]
+                                for cmd in cmds:
+                                    stdout, stderr = _run_command(cmd)
+                                if stdout.decode("utf-8") == "Hello, world!":
+                                    success, err, mark_fraction = True, "", 1.0
+                                return success, err, mark_fraction
+                            """
+                            )
+
+        hook_code_2 = dedent("""\
+                             def check_answer(user_answer):
+                                success = False
+                                err = "Incorrect Answer"
+                                mark_fraction = 0.0
+                                if 'System.out.print("Hello, world!");' in user_answer:
+                                    success, err, mark_fraction = True, "", 0.5
+                                return success, err, mark_fraction
+                            """
+                            )
+
+
+        test_case_data = [{"test_case_type": "hooktestcase",
+                           "hook_code": hook_code_1, 'weight': 1.0},
+                          {"test_case_type": "hooktestcase",
+                           "hook_code": hook_code_2, 'weight': 1.0},
+                          ]
+        kwargs = {
+                  'metadata': {
+                    'user_answer': user_answer,
+                    'file_paths': self.file_paths,
+                    'partial_grading': True,
+                    'language': 'java'
+                    },
+                    'test_case_data': test_case_data,
+                  }
+
+        # When
+        grader = Grader(self.in_dir)
+        result = grader.evaluate(kwargs)
+
+        # Then
+        self.assertTrue(result.get('success'))
+        self.assertEqual(result.get("weight"), 1.5)
+        
+    def test_infinite_loop(self):
+        # Given
+        user_answer = dedent("""\
+                             class Test
+                             {public static void main(String[] args){
+                             while(0==0)
+                            {
+                            System.out.print("a");}
+                            }}""")
+
+        hook_code = dedent("""\
+                            def check_answer(user_answer):
+                                with open("Test.java", "w+") as f:
```

Comment:
Do you really need such a complex test?  All you have is either the test code returning different values depending on the answer.  How that is actually evaluated is a detail.  While the test does serve as an interesting example I am afraid it may make maintaining the tests difficult.  Many of the other tests are also similar.